src/web/extension.ts

```diff
@@ -4,9 +4,17 @@
  * ------------------------------------------------------------------------------------------ */
 
 import * as vscode from "vscode";
+import TelemetryReporter from "@vscode/extension-telemetry";
+import { AI_KEY } from '../client/constants';
+import { randomUUID } from "crypto";
+
+let _telemetry: TelemetryReporter;
 
 export function activate(context: vscode.ExtensionContext): void {
     console.log("Activated web extension!"); // sample code for testing the webExtension
+    // setup telemetry
+    _telemetry = new TelemetryReporter(context.extension.id, context.extension.packageJSON.version, AI_KEY);
```

Comment:
please use the AI key for development instance since this extension is not (hopefully) public yet.